readiness-tool/popup.js

```diff
@@ -0,0 +1,261 @@
+  notSupportedAnalytics.appendChild(
+    makeList(detectedApps.notSupported.analytics));
+  notSupportedOther.appendChild(makeList(detectedApps.notSupported.other));
+}
+
+/**
+ * Splits all detected apps into 'supported' and 'not supported'
+ * @param {Object} apps - All possible services that could exist on the page
+ * @param {String} htmlString - String containing all HTML on the page
+ * @return {Object} 
+ */
+function specifySupport(apps, htmlString) {
+  const obj = apps;
+  let foundThis = {
+    'supported': {
+      'ads': [],
+      'analytics': [],
+    },
+    'notSupported': {
+      'ads': [],
+      'analytics': [],
+      'other': [],
+    },
+  };
+
+  // for all the app objects in the apps.JSON file
+  Object.keys(obj).forEach(function(key) {
+    let val = obj[key];
+    // If object has a 'script' key
+    if (val.script != null) {
+      // Sometimes val.script contains an array of regular expressions
+      if (typeof val.script == 'object') {
+        Object.keys(val.script).forEach(function(x) {
+          addToDict(val.script[x], htmlString, foundThis, key, val.cats);
+        });
+      } else {
+        addToDict(val.script, htmlString, foundThis, key, val.cats);
+      }
+    }
+  });
+
+  return foundThis;
+}
+
+
+/**
+ * Pushes keys to the supported or not supported list of the object 'found this'
+ * @param {String} tempScript - String representation of regular expression
+ * @param {!String} htmlString - String containing all HTML on the page
+ * @param {Object} foundThis - Object separating the 3P services by support
+ * @param {String} key - name of third party service
+ * @param {String} category - the category that the key belongs to
+ */
+function addToDict(tempScript, htmlString, foundThis, key, category) {
+  tempScript = tempScript.split('\\;');
+  let regX = new RegExp(tempScript[0]);
+  if (doesRegexExist(regX, htmlString)) {
+    if (isKeyUnique(foundThis, key)) {
+      if (isSupported(key)) {
+        findCategory(category, foundThis.supported, key);
+      } else {
+        findCategory(category, foundThis.notSupported, key);
+      }
+    }
+  }
+}
+
+/**
+ * Checks to see if key is unique within the object
+ * @param {Object} obj - Object separating the 3p services by support
+ * @param {String} key - name of third party service
+ * @return {boolean} 
+ */
+function isKeyUnique(obj, key) {
+  let truthValue = obj.supported.ads.indexOf(key) +
+      obj.supported.analytics.indexOf(key) + obj.notSupported.ads.indexOf(key)+
+      obj.notSupported.analytics.indexOf(key);
+  return truthValue == -4;
+}
+
+/**
+ * Figures out thae category the key belongs to (Ads or Analytics)
+ * @param {String} category - number representing the category of each item
+ * @param {Object} objectList - The found supported/unsupported 3p services
+ * @param {String} key - name of third party service
+ */
+function findCategory(category, objectList, key) {
```

Comment:
I think this function should also be inlined since:
- It's only used in one place
- It doesn't really make sense outside of the context of `addToDict`